src/applications/applications.service.ts

```diff
@@ -102,27 +107,37 @@ export class ApplicationsService {
         application: CreateApplicationDto,
         userId: number,
     ): Promise<ReturnApplicationDto> {
-        let job = plainToInstance(Job, application);
+        let jobEntity = Job.fromApplication(application);
         let applicationEntity = new Application();
         const queryRunner = this.dataSource.createQueryRunner();
         await queryRunner.connect();
         await queryRunner.startTransaction();
         let returnApplicationDto: ReturnApplicationDto;
+
         try {
-            job = await queryRunner.manager.save(job);
-            this.logger.debug(`Job created successfully: ${job.company}`);
-            applicationEntity.jobId = job.id;
+            // Save the Job entity
+            jobEntity = await queryRunner.manager.save(jobEntity);
+            this.logger.debug(`Job created successfully: ${jobEntity.company}`);
+
+            // Create application with the saved job ID
+            applicationEntity.jobId = jobEntity.id;
             applicationEntity.userId = userId;
-            applicationEntity.status = application.status|| ApplicationStatus.Apply;
-            applicationEntity.appliedDate = application.appliedDate || new Date();
+            applicationEntity.status =
+                application.status ?? ApplicationStatus.Apply;
+            applicationEntity.appliedDate =
+                application.appliedDate ?? new Date();
+
             applicationEntity =
                 await queryRunner.manager.save(applicationEntity);
             this.logger.debug(
-                `Application created successfully for Application ID: ${applicationEntity.id} for Company ${job.company}`,
+                `Application created successfully for Application ID: ${applicationEntity.id} for Company ${jobEntity.company}`,
             );
+
             await queryRunner.commitTransaction();
+
+            const jobDto = plainToInstance(Job, jobEntity);
             returnApplicationDto = await this.getReturnApplicationDto(
-                job,
+                jobDto,
```

Comment:
`this.getReturnApplicationDto(job: Job, applicationEntity: Application)` 
We don't need to convert in line 138. We can send the jobEntity as is. 
```suggestion
returnApplicationDto = await this.getReturnApplicationDto(jobEntity, applicationEntity);
```